### PR TITLE
Sofia Willebrand

### PR DESCRIPTION
**Pull request by sofiawillebrand**
Link to my deployed project:
https://what-do-i-have-to-do.netlify.app/

My reflections on how this week's project turned out:
I am very pleased with how much I accomplished this week (blue, red and black), but I do feel a bit behind when it comes to the styling. 

Things I'd like to have clarified or explained in more detail:
Is it okay to mix React useState and Redux Store? 
How should one handle Styled components? Is it okay to add them in one single file, and have a style-library?

### CODE REVIEW

File: code/src/App.js [hunk 1]
```diff
@@ -1,9 +1,72 @@
...
+  height: 100vh;
+  font-family: "rubik";
+  margin-bottom: 150px;
+`;
+
+// by convention always called reducer
+const reducer = combineReducers({
+  // all the slices are specified here
+  todos: todos.reducer,
+});
```
Comment (Skrosen) [inline code/src/App.js:33]: Nice with the comments to explain, however I don't think you need to combine reducers when there is only one reducer.

File: code/src/App.js [hunk 2]
```diff
@@ -1,9 +1,72 @@
...
-      Find me in src/app.js!
-    </div>
-  )
-}
+    <Provider store={store}>
+      <GlobalStyle />
+      <Wrapper>
+        <Header />
+        {!newAddTodo && <OpenTodo setNewAddTodo={setNewAddTodo} />}
+        {newAddTodo && <AddTodo setNewAddTodo={setNewAddTodo} />}
```
Comment (Skrosen) [inline code/src/App.js:67]: Nice solution!

File: code/src/components/AddProject.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
+import React from "react";
+import Select from "react-select";
+
+const options = [
+  { value: "prio1", label: "Prio 1" },
+  { value: "prio2", label: "Prio 2" },
+  { value: "prio3", label: "Prio 3" },
+];
```
Comment (Skrosen) [inline code/src/components/AddProject.js:8]: I like this option!

File: code/src/components/AddTag.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+  borderRadius: 0,
+  colors: {
+    ...theme.colors,
+    primary25: "purple",
+  },
+});
+
+const animatedComponents = makeAnimated();
+
+const AddTag = ({ setNewTags, newTags }) => {
```
Comment (Skrosen) [inline code/src/components/AddTag.js:37]: I really like the tag function that you added!

File: code/src/components/AddProject.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+const theme = (theme) => ({
+  ...theme,
+  borderRadius: 0,
+  colors: {
+    ...theme.colors,
+    primary25: "purple",
+  },
+});
+
+const AddTag = ({ setNewProject, newLabel, setNewLabel }) => {
```
Comment (Skrosen) [inline code/src/components/AddProject.js]: This is called the same as the tag-function, is there a reason why?

File: code/src/components/AddTodo.js [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+  const [newProject, setNewProject] = useState("");
+  const [newDate, setNewDate] = useState(new Date());
+  const [newTags, setNewTags] = useState([]);
+  const [newLabel, setNewLabel] = useState("");
+
+  const dispatch = useDispatch();
+
+  const onAddTodo = () => {
+    if (newTodo === "") {
+      alert("Hey there, there is no todo! Fill in your todo");
```
Comment (Skrosen) [inline code/src/components/AddTodo.js:77]: Nice alert to interact with the user!

**Review by Skrosen — COMMENTED** [review]
Really nice work! Your code looks very well written and your comments are clear and helpful! You have divided your code in a structured way which makes it easy to navigate. Very well done!!

File: code/src/components/AddProject.js (hunk 5 above)
Comment (sofiawillebrand) [reply code/src/components/AddProject.js]: Oh no, that is a typo! Thank you for pointing it out.

**Pull request closed without merging**